lib/cereal/cereal.go

```diff
@@ -769,29 +771,24 @@ func (m *Manager) processWorkflow(ctx context.Context, workflowNames []string) b
 		if err != nil {
 			logctx.WithError(err).Error("failed to complete workflow")
 		}
-
 		s.End("complete")
 	} else if decision.continuing {
 		s.Begin("enqueue_task")
 		for _, t := range decision.tasks {
 			err := completer.EnqueueTask(&t, backend.TaskEnqueueOpts{})
 			if err != nil {
-				// TODO(ssd) 2019-05-15: What do we
-				// want to do here? I think we need to
-				// rollback and have the workflow run
-				// again.
 				logrus.WithError(err).Error("failed to enqueue task!")
+				return true
```

Comment:
We've defer'd a close on the completer, so this should end up rolling back our transaction and the workflow step should get retried.  This assumes the most likely errors here are pg errors.